File: frontend/vue-vben-admin/apps/web-antd/src/api/system/login-log.ts
import { requestClient } from '#/api/request';

export interface LoginLogItem {
  createdAt: string;
  id: string;
  ipAddress?: null | string;
  isSuccess: boolean;
  message: string;
  realName?: null | string;
  userAgent?: null | string;
  userId?: null | string;
  userName: string;
}

export interface LoginLogListParams {
  endCreatedAt?: string;
  isSuccess?: boolean;
  page?: number;
  pageSize?: number;
  startCreatedAt?: string;
  userName?: string;
}

export interface LoginLogListResult {
  items: LoginLogItem[];
  total: number;
}

export async function getLoginLogListApi(params: LoginLogListParams) {
  return requestClient.get<LoginLogListResult>('/system/login-log/list', {
    params,
  });
}

File: src/MiniAdmin.Domain/Entities/LoginLog.cs
namespace MiniAdmin.Domain.Entities;

public sealed class LoginLog
{
    public Guid Id { get; set; }

    public Guid? UserId { get; set; }

    public string UserName { get; set; } = string.Empty;

    public string? RealName { get; set; }

    public string? IpAddress { get; set; }

    public string? UserAgent { get; set; }

    public bool IsSuccess { get; set; }

    public string Message { get; set; } = string.Empty;

    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
}
